Test/Algorithms/Trainers/LogisticRegression.cpp

```diff
@@ -0,0 +1,157 @@
+#define BOOST_TEST_MODULE TRAINERS_LDA
+#include <boost/test/unit_test.hpp>
+#include <boost/test/floating_point_comparison.hpp>
+
+#include <shark/Algorithms/Trainers/LogisticRegression.h>
+#include <shark/Data/DataDistribution.h>
+#include <shark/ObjectiveFunctions/Loss/CrossEntropy.h>
+#include <shark/ObjectiveFunctions/ErrorFunction.h>
+using namespace shark;
+
+BOOST_AUTO_TEST_SUITE (Algorithms_Trainers_Logistic_Regression)
+
+BOOST_AUTO_TEST_CASE( LogReg_Binary_NoLone){
+	PamiToy problem;
+	ClassificationDataset const& dataset = problem.generateDataset(50);
+	
+	LogisticRegression<> trainer;
+	BOOST_CHECK_EQUAL(trainer.lambda1(),0.0);
+	BOOST_CHECK_EQUAL(trainer.lambda2(),0.0);
+	BOOST_CHECK_EQUAL(trainer.parameterVector()(0),0.0);
+	BOOST_CHECK_EQUAL(trainer.parameterVector()(1),0.0);
+	BOOST_CHECK_EQUAL(trainer.parameterVector().size(),2);
+	BOOST_CHECK_EQUAL(trainer.numberOfParameters(),2);
+	trainer.setLambda2(0.1);
+	BOOST_CHECK_EQUAL(trainer.lambda1(),0.0);
+	BOOST_CHECK_EQUAL(trainer.lambda2(),0.1);
+	BOOST_CHECK_EQUAL(trainer.parameterVector()(0),0.0);
+	BOOST_CHECK_EQUAL(trainer.parameterVector()(1),0.1);
+	BOOST_CHECK_EQUAL(trainer.accuracy(), 1.e-8);
+	LogisticRegression<>::ModelType model;
+	trainer.train(model,dataset);
+	
+	BOOST_CHECK_EQUAL(model.numberOfParameters(),11);
+	
+	//check gradient
+	{
+		CrossEntropy loss;
+		ErrorFunction error(dataset, &model.decisionFunction(),&loss);
+		RealVector derivative;
+		error.evalDerivative(model.parameterVector(),derivative);
+		RealVector penalty = 0.1 * model.parameterVector();
+		penalty(10) = 0;
```

Comment:
Is this the bias? If so, is it possible to obtain the linear part of the function directly from the model, not indirectly through the parameter vector?